### ISSUE TITLE
[Bug]: Eufy Indoor Cam 2k alarm state issue

### ISSUE THREAD

**Issue by njuart** [homebridge-eufy-security/plugin#38]
### What happened?

Hi, I have both Homebase 2 and Eufy Indoor Cam 2k. When I change alarm state for example from Away to Home in HomeKit, I see that alarm state of Indoor Cam doesn't change, it changes only after I manually open Eufy App and check state there. After this state of Indoor Cam in HomeKit changes immediately. With Homebase 2 I don't have this problem, it's alarm state changes at once. Any ideas? Thx

### Device Type

Indoor Camera (Supported)

### Plugin Version

beta (Best Effort)

### HomeBridge Version

v1.4.x (Supported)

### NodeJS Version

v16 (Supported)

### Operating System

Raspbian 10+ (Supported)

**Comment by thieren** [comment]
As discussed in #39 

> Yes it occurs every time, but on earlier betas it was ok. When I arm it doesn't show it's armed at all, after I open Eufy app and refresh, it becomes armed. And when I disarm it hangs in arming state. Honestly I'm not sure if it hangs indefinitely, maybe a couple of times it was disarmed by itself, maybe after few minutes or so.

@njuart Can you specify on which version of the plugin it was working?
Also can you supply log output with debug enabled? Your config without the personal information would also be helpful. Thanks!

**Comment by njuart** [comment]
> As discussed in #39
> 
> > Yes it occurs every time, but on earlier betas it was ok. When I arm it doesn't show it's armed at all, after I open Eufy app and refresh, it becomes armed. And when I disarm it hangs in arming state. Honestly I'm not sure if it hangs indefinitely, maybe a couple of times it was disarmed by itself, maybe after few minutes or so.
> 
> @njuart Can you specify on which version of the plugin it was working? Also can you supply log output with debug enabled? Your config without the personal information would also be helpful. Thanks!

Sorry, but not sure I can recall on which beta exactly it was working. Here's what log shows:

<img width="960" alt="image" src="https://user-images.githubusercontent.com/21228802/170882325-de700f1f-bbe2-448e-a6ef-ddcc4e4a5f5b.png">

1. I've armed Camera to Away from Home app
2. Saw infinite arming state
3. (Red arrow pointer) It's when I've opened Eufy app and refreshed Security tab
4. After I got back to Home app state was set to Away

**Comment by lenoxys** [comment]
Hello, I've the same issue on my setup. When arming we send a request to arm to the camera then we wait to receive the arming notification which can happend minutes after the ask. I don't see anything we can implement in order to force it. This issue is only related to camera operating their own alarm system. I don't have this issue with the homebase.

**Comment by njuart** [comment]
> Hello, I've the same issue on my setup. When arming we send a request to arm to the camera then we wait to receive the arming notification which can happend minutes after the ask. I don't see anything we can implement in order to force it. This issue is only related to camera operating their own alarm system. I don't have this issue with the homebase.

Yes, with Homebase it's fine. But I'm pretty sure it was working fine some time ago.

**Comment by thieren** [comment]
> Yes, with Homebase it's fine. But I'm pretty sure it was working fine some time ago.

Well since I don't own a solo camera I have difficulties to help here.
From looking through the code there is not much going on. We're doing exactly one thing and that is to give the command to the eufy-security-client package and then wait. I don't believe that this was changed much in the earlier versions but didn't have a closer look.
Honestly there are a myriad of things that could've changed that could cause your issue (software updates from eufy, maybe an update in the eufy-security-client dependency, ...) but as mentioned I don't believe that the cause is in this plugin.

You can revert to earlier version of the plugin and test if it has another behavior. Maybe then I can find a difference in the code bases.

**Comment by thieren** [comment]
I've added a log message that should be triggered if the mode change isn't done.

It's part of 2.0.1-beta.2. Until it's published you could download it directly from the repo.

Then there will be exactly one retry. But as stated before in #39 and from the information given I don't believe that the retry will do any good.

Please report if this helps.

**Comment by njuart** [comment]
> I've added a log message that should be triggered if the mode change isn't done.
> 
> 
> 
> It's part of 2.0.1-beta.2. Until it's published you could download it directly from the repo.
> 
> 
> 
> Then there will be exactly one retry. But as stated before in #39 and from the information given I don't believe that the retry will do any good.
> 
> 
> 
> Please report if this helps.

Didn't help unfortunately. There is a timeout =( Even after retry.

**Comment by thieren** [comment]
> Didn't help unfortunately. There is a timeout =( Even after retry.

Yeah, I expected that really.

I suppose there is a power management setting in the eufy app. Could you test whether different settings have different results in the homebridge behavior?

**Comment by njuart** [comment]
Hmm, didn't find any in Eufy app. Maybe because my Indoor 2k is wired.

**Comment by njuart** [comment]
> Ok, another try.
> 
> 
> 
> I've hacked a eufy-test-client application which shall help me debug some issues in the future. For now there are only two options to change guard mode.
> 
> 
> 
> You can find it here: https://github.com/thieren/eufy-test-client
> 
> It's just quick and dirty, so don't expect too much.
> 
> 
> 
> Can you try something with this please?
> 
> 
> 
> 1. If you use the standard 'change guard mode' option, you should get the same result as with the plugin.
> 
> 2. The alternative option is a clutch at a straw to get this working for you.
> 
> 
> 
> Please report back, if this does anything for you.

Wow, thx. Not sure if I'll be near computer today, so probably tmrw. Maybe @lenoxys can assist you also, as he reported the same issue for him a bit a above in this thread.

**Comment by njuart** [comment]
@thieren Hey, again. Luckily managed to do it today and got the log. So:

1. I've changed Camera guard mode to 0 (AWAY) via your client
2. After few minutes opened Home app, nothing changed there (as expected unfortunately)
3. Opened Eufy app, Camera guard mode there was already AWAY, which I suppose means that it was successfully changed via your client
4. After it opened Home app again and status there was also AWAY this time

Here's the log
[eufylog_1654003655751.txt](https://github.com/homebridge-eufy-security/plugin/files/8805716/eufylog_1654003655751.txt)

**Comment by thieren** [comment]
Did you try the alternative change guard mode option also? According to the log you just used one.

**Comment by njuart** [comment]
> Did you try the alternative change guard mode option also? According to the log you just used one.

I wanted to, but got disconnected, and for some reason I can't connect with your client again.

**Comment by thieren** [comment]
> I wanted to, but got disconnected, and for some reason I can't connect with your client again.

As I stated: just a quick and dirty solution. Can you write to me on Discord? Maybe we can work on this. But I think this is out of scope for this issue.

**Comment by thieren** [comment]
Hey @njuart 

did update my application to handle 2fa and captchas. You should now be able to login again I think.
Let me know if it helps.

**Comment by njuart** [comment]
> Hey @njuart 
> 
> 
> 
> did update my application to handle 2fa and captchas. You should now be able to login again I think.
> 
> Let me know if it helps.

I didn't update yet, but yesterday I've disabled 2FA to be able to login, first time it worked, but had no luck after. Will try a new version and come back. Thx!

**Comment by njuart** [comment]
@thieren Just tried new client, but I don't see where I should put 2FA code, even though I get it on my mail while initialization.

**Comment by thieren** [comment]
Confirmed with @njuart that this is not a plugin issue, but rather with eufy-security-client
Maybe @bropat can do something about this.

Closed.

**Comment by thieren** [comment]
Should be fixed with latest beta.
